src/profiles.jl

```diff
@@ -512,36 +512,154 @@ function add_metadata!(commp::CommunityProfile, samplecol::Symbol, md; overwrite
     return nothing
 end
 
+"""
+    set!(commp::CommunityProfile, sample::AbstractString, prop::Symbol, val)
+
+Update or insert a value `val` to the `sample` in the CommunityProfile `commp` using a Symbol `prop`. 
+
+Examples
+≡≡≡≡≡≡≡≡≡≡
+
+```jldoctest
+julia> set!(comm, "sample1", :something, 1.0)
+
+julia> first(metadata(comm))[:something]
+1.0
+ ```
+"""
 function set!(commp::CommunityProfile, sample::AbstractString, prop::Symbol, val)
     sample = samples(commp, sample)
     prop in _restricted_fields(sample) && error("Cannot set! $prop for $(typeof(sample)).")
     set!(sample.metadata, prop, val)
     return sample
 end
 
+"""
+    unset!(commp::CommunityProfile, sample::AbstractString, prop::Symbol)
+
+Delete a `sample` from CommunityProfile `commp` using the Symbol `prop` if it exists, or throw an error otherwise.
```

Comment:
`unset!` will not throw an error if `prop` doesn't exist. Please take a look [here](https://github.com/andyferris/Dictionaries.jl#accessing-dictionaries), and update docstrings accordingly.

It would also be good to add cross references that mentions methods available if you do or don't want errors to be thrown? Eg, `unset!` will succeed whether or not `prop` exists, and `delete!` will error if it doesn't. `set!` is like `insert!`, except that it will overwrite the value if it exists. You can use `@ref` on methods to link them, eg

```
If you want an error to be thrown if the value does not exist, use [`delete!`](@ref).
```